activesupport/lib/active_support/rescuable.rb

```diff
@@ -99,31 +137,26 @@ def handler_for_rescue(exception)
         # Note that this tolerance only matters if the exception was given as
         # a string, otherwise a NameError will be raised by the interpreter
         # itself when rescue_from CONSTANT is executed.
-        klass = self.class.const_get(klass_name) rescue nil
-        klass ||= (klass_name.constantize rescue nil)
-        klass === exception if klass
-      end
-
-      case rescuer
-      when Symbol
-        method(rescuer)
-      when Proc
-        if rescuer.arity == 0
-          Proc.new { instance_exec(&rescuer) }
-        else
-          Proc.new { |_exception| instance_exec(_exception, &rescuer) }
+        def constantize_rescue_handler_class_name(class_name)
+          begin
+            const_get class_name
+          rescue
+            begin
+              class_name.constantize
+            rescue
+              nil
+            end
+          end
```

Comment:
Looking back to the origin (https://github.com/rails/rails/commit/964dfc15572d7c10771c81ac3cbfb455dd5e378e), that seems accidental at best. In that case, you'd just use the constant itself, not a string referencing it.